packages/core/test/test_util/common_tests.js

```diff
@@ -0,0 +1,131 @@
+/*
+ * (c) Copyright IBM Corp. 2021
+ * (c) Copyright Instana Inc. and contributors 2021
+ */
+
+'use strict';
+
+const expectExactlyOneMatching = require('./expectExactlyOneMatching');
+const constants = require('../../src/index').tracing.constants;
+
+/**
+ * @typedef {import('../../src/tracing/cls').InstanaBaseSpan} InstanaBaseSpan
+ */
+
+/**
+ * @param {Chai.ExpectStatic} expect
+ * @param {Array.<InstanaBaseSpan>} spans
+ * @param {string} apiPath
+ * @param {string} pid
+ * @param {Array.<(span: InstanaBaseSpan) => void>} [extraTests]
+ * @returns
+ */
+exports.verifyHttpEntry = function verifyHttpEntry(expect, spans, apiPath, pid, extraTests) {
+  /** @type {Array.<(span: InstanaBaseSpan) => void>} */
+  const tests = [
+    (/** @type {InstanaBaseSpan} */ span) => {
+      expect(span.p).to.not.exist;
```

Comment:
This is only true for root entry spans. If this method is meant to check for HTTP root entry spans, it should be named accordingly (verifyHttpRootEntry).

Alternatively: We could add an optional parentSpan to the arguments. If it is present, we can test

```
expect(span.t).to.equal(parentSpan.t);
expect(span.p).to.equal(parentSpan.s);
```

If it is not present, we test

```
expect(span.p).to.not.exist;
```

The optional parentSpan argument would be a bit ugly with the current positional arguments, so this might only be a good idea if we switch to named arguments (see above).

Then again, maybe it is better to add explicit methods for both cases, like verifyHttpRootEntry and verifyHttpNonRootEntry (which might both delegate to the same method which then takes an optional parent span).

Reply:
My thinking was towards parent HTTP entry, so I think it makes sense to rename it, as you said.
This is the most common scenario, as we discussed in the Power Point presentation I presented to you: HTTP Server ENTRY -> INSTRUMENTED_MODULE EXIT -> INSTRUMENTED_MODULE ENTRY -> HTTP Client EXIT.

I suggest to keep it as is for now (no optional argument, renaming the function properly), and we discuss again when the next wave of refactoring comes up.